ext/websocket/lib.rs

```diff
@@ -243,17 +245,25 @@ where
   let client =
     fastwebsockets::handshake::client(&LocalExecutor, request, socket);
 
-  let (stream, response): (WebSocket<Upgraded>, Response<Body>) =
-    if let Some(cancel_resource) = cancel_resource {
-      client.or_cancel(cancel_resource.0.to_owned()).await?
-    } else {
-      client.await
-    }
-    .map_err(|err| {
-      DomExceptionNetworkError::new(&format!(
-        "failed to connect to WebSocket: {err}"
-      ))
-    })?;
+  let (upgraded, response) = if let Some(cancel_resource) = cancel_resource {
+    client.or_cancel(cancel_resource.0.to_owned()).await?
+  } else {
+    client.await
+  }
+  .map_err(|err| {
+    DomExceptionNetworkError::new(&format!(
+      "failed to connect to WebSocket: {err}"
+    ))
+  })?;
+
+  let Parts { io, read_buf, .. } = upgraded
+    .into_inner()
+    .downcast::<MaybeTlsStream<TcpStream>>()
+    .unwrap();
```

Comment:
```suggestion
```
Don't downcast. Instead make the enum variant a `WsStreamKind::Tungstenite(MaybeTlsStream<Upgraded>)`